plugins/obs-filters/noise-suppress-filter.c

```diff
@@ -325,29 +356,44 @@ static void noise_suppress_update(void *data, obs_data_t *s)
 
 	ng->use_nvafx = ng->nvafx_enabled &&
 			strcmp(method, S_METHOD_NVAFX) == 0;
-#ifdef LIBNVAFX_ENABLED
-	ng->intensity_ratio = (float)obs_data_get_double(s, S_NVAFX_INTENSITY);
-#endif
+
 	/* Process 10 millisecond segments to keep latency low */
 	/* Also RNNoise only supports buffers of this exact size. */
 	/* At 48kHz, NVAFX processes 480 samples which corresponds to 10 ms.*/
 	ng->frames = frames;
 	ng->channels = channels;
 
+#ifdef LIBNVAFX_ENABLED
+	ng->intensity_ratio = (float)obs_data_get_double(s, S_NVAFX_INTENSITY);
+	if (ng->nvafx_initialized && ng->use_nvafx) {
+		int err;
+		for (int i = 0; i < MAX_PREPROC_CHANNELS; i++) {
+			err = NvAFX_SetFloat(
+				ng->handle[i],
+				NVAFX_PARAM_DENOISER_INTENSITY_RATIO,
+				ng->intensity_ratio);
+			if (err != NVAFX_STATUS_SUCCESS) {
+				do_log(LOG_ERROR,
+				       "NvAFX_SetFloat(Intensity Ratio: %d) failed, error %i",
```

Comment:
Is there a reason you're using `%d` here instead of `%f` the float?

Reply:
yeah i goofed. Thx